flocker/restapi/test/test_infrastructure.py

```diff
@@ -462,6 +479,21 @@ def test_noBodyPOST(self, logger):
         """
         self.assertNoDecodeLogged(logger, b"POST", b"/foo/ignore_body")
 
+    @validateLogging(_assertRequestLogged(b"/foo/ignore_body", b"POST"))
+    def test_noBodyPOSTnotJSON(self, logger):
+        """
+        A non-I{JSON} request is body is ignored for methods with
+        C{{ignore_body}} set to C{{True}}.
+        """
+        request = dummyRequest(
+            b"POST", b"/foo/ignore_body",
+            Headers({b"content-type": [b"x-application/garbage"]}),
+            b"this is some non-JSON garbage! {} {}")
+
+        app = self.Application(logger, None)
+        render(app.app.resource(), request)
+        self.assertEqual({}, app.kwargs)
```

Comment:
This would be clearer if `assertNoDecodeLogged` was modified to have the signature `assertNoDecodeLogged(self, logger, method, path=b"/foo/bar", content_type=b"application/json", objects={"foo": "bar", "baz": ["quux"]})`. This would clarify this test by using the same validation function as the previous related tests.